spras/config/config.py

```diff
@@ -42,39 +43,41 @@ def init_from_file(filepath):
     try:
         with open(filepath, 'r') as yaml_file:
             config_dict = yaml.safe_load(yaml_file)
-    except FileNotFoundError:
-        print(f"Error: The specified config '{filepath}' could not be found.")
-        return False
+    except FileNotFoundError as e:
+        raise RuntimeError(f"Error: The specified config '{filepath}' could not be found.") from e
     except yaml.YAMLError as e:
-        print(f"Error: Failed to parse config '{filepath}': {e}")
-        return False
+        raise RuntimeError(f"Error: Failed to parse config '{filepath}'") from e
 
     # And finally, initialize
     config = Config(config_dict)
 
 
 class Config:
-    def __init__(self, raw_config):
-        # Since process_config winds up modifying the raw_config passed to it as a side effect,
-        # we'll make a deep copy here to guarantee we don't break anything. This preserves the
-        # config as it's given to the Snakefile by Snakemake
+    def __init__(self, raw_config: dict[str, Any]):
+        # Since snakemake provides an empty config, we provide this
+        # wrapper error first before passing validation to pydantic.
+        if raw_config == {}:
+            raise ValueError("Config file cannot be empty. Use --configfile <filename> to set a config file.")
+
+        parsed_raw_config = RawConfig.model_validate(raw_config)
```

Comment:
I assume this no longer modifies `raw_config` which is why we no longer make a deep copy?